build-scripts/package-tests.cake

```diff
@@ -0,0 +1,382 @@
+// Representation of a single test to be run against a pre-built package.
+// Each test has a Level, with the following values defined...
+//  0 Do not run - used for temporarily disabling a test
+//  1 Run for all CI tests - that is every time we test packages
+//  2 Run only on PRs, dev builds and when publishing
+//  3 Run only when publishing
+public struct PackageTest
+{
+	public int Level;
+	public string Description;
+	public string Arguments;
+	public ExpectedResult ExpectedResult;
+	public string[] ExtensionsNeeded;
+	
+	public PackageTest(int level, string description, string arguments, ExpectedResult expectedResult, params string[] extensionsNeeded)
+	{
+		Level = level;
+		Description = description;
+		Arguments = arguments;
+		ExpectedResult = expectedResult;
+		ExtensionsNeeded = extensionsNeeded;
+	}
+}
+
+// Abstract base for all package testers. 
+public abstract class PackageTester
+{
+    protected BuildParameters _parameters;
+    protected ICakeContext _context;
+
+    public PackageTester(BuildParameters parameters)
+    {
+        _parameters = parameters;
+        _context = parameters.Context;
+
+        PackageTests = new List<PackageTest>();
+    }
+
+    protected abstract string PackageName { get; }
+    protected abstract FilePath PackageUnderTest { get; }
+    protected abstract string PackageTestDirectory { get; }
+    protected abstract string PackageTestBinDirectory { get; }
+    protected abstract string ExtensionInstallDirectory { get; }
```

Comment:
Is any of the extension functionality in this class used yet? I've been trying to follow through where it's set, but can't find anywhere that sets any of the variables. Again, if not, could we pull it out and simplify for now?